frontend/public/components/modals/remove-volume-modal.tsx

```diff
@@ -0,0 +1,90 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory';
+import { history, resourcePathFromModel } from '../utils';
+import { getVolumeType, K8sKind, k8sPatch, K8sResourceKind } from '../../module/k8s/';
+import { RowVolumeData } from '../volumes-table';
+
+export const RemoveVolumeModal: React.FC<RemoveVolumeModalProps> = (props) => {
+  const [inProgress, setInProgress] = React.useState(false);
+  const [errorMessage, setErrorMessage] = React.useState('');
+
+  const getRemoveVolumePatch = (resource, rowVolumeData: RowVolumeData) => {
+    const location = resource.kind === 'Pod' ? 'spec' : 'spec.template.spec';
+    const dirLocation = resource.kind === 'Pod' ? '/spec' : '/spec/template/spec';
+    const containers = _.get(resource, `${location}.containers`, []);
```

Comment:
I may have tried to be too clever here to include removing volumes from Pods - I don't think the API let's you do this for any volumes.  I probably need to state 'Volumes cannot be removed from Pods' on the modal window.

Reply:
@spadgett - I thinik I have 2 options for Pod resources: 1) Don't show the kebab at all if the only action is `Remove Volume` or 2) Show an error screen instead of a confirmation screen when someone tries to remove a volume from a Pod.
@bmignano might have some better opinion from the UX perspective whether we 1) should show an error page saying you cannot do that or if we 2) should just **not** even give them a choice and they wonder 'why don't I have the kebab option here?'

Reply:
@zherman0 Good question! I think a possible third option is to allow users to open the kebab but disable that action. I found this similar situation in the console today which I think we could model this behavior after. The tooltip could read something like "Volumes cannot be removed from pods" Thoughts?

![Screen Shot 2019-07-08 at 1 40 01 PM](https://user-images.githubusercontent.com/40238481/60830804-069aab00-a186-11e9-8ff3-736738f0c561.png)

Reply:
@bmignano - That looks great.  I would most prefer to have consistent behaviour across the console so if someone else is doing that, I would like to follow that pattern. Thanks!

Reply:
Since you can never remove a volume for any pod, I wouldn't show the kebab. If all actions will always be disabled, we should hide it imo.

Reply:
What about if we every do have kebab actions on pods?  Would it be good to already have the kebab there? I am fine either way, just want to help out the next guy who might need an action added to the kebab.

Reply:
I'm not sure what action we'd ever have here for pods since the pod spec is immutable. You can't edit, add, or remove volumes. Let's keep things simple until we need a kebab.